packages/nest/src/open-feature.module.ts

```diff
@@ -0,0 +1,45 @@
+import { DynamicModule, Module, FactoryProvider as NestValueProvider } from '@nestjs/common';
+import { Client, OpenFeature, Provider } from '@openfeature/server-sdk';
+
+@Module({})
+export class OpenFeatureModule {
+  static forRoot(options: OpenFeatureModuleOptions): DynamicModule {
+    const clientValueProviders: NestValueProvider<Client>[] = [];
+
```

Comment:
Not sure, as the SDKs themselves also do not throw, but rather use the No-Op Provider.
If we throw or enforce a provider, we would behave differently than the SDKs I would say.
But ya, using the No-Op Provider does not make much sense but still I would rather behave like the SDK.

Reply:
I see.

What do you think about configuring No-Op as default provider if no named provider or default provider is passed?
It might not throw at this point, but if the provider injection decorator is being used the app will break anyway

Reply:
> What do you think about configuring No-Op as default provider if no named provider or default provider is passed?

Yes thats good. We should just always inject the default provider regardless of what it is set to.
Added that in the last commit.